utilities/rpp-unittests/SOW3_HOST/BatchPD_host_pln3.cpp

```diff
@@ -0,0 +1,921 @@
+
+    char funcType[1000] = {"BatchPD_HOST_PLN3"};
+
+    if (outputFormatToggle == 0)
+    {
+        strcat(funcType, "_toPLN3");
+    }
+    else if (outputFormatToggle == 1)
+    {
+        strcat(funcType, "_toPKD3");
+    }
+
+    char funcName[1000];
+    switch (test_case)
+    {
+    case 1:
+        strcpy(funcName, "rotate");
+        break;
+    case 2:
+        strcpy(funcName, "resize");
+        break;
+    case 3:
+        strcpy(funcName, "resize_crop");
+        break;
+    case 4:
+        strcpy(funcName, "resize_crop_mirror");
+        break;
+    case 5:
+        strcpy(funcName, "crop");
+        break;
+    case 6:
+        strcpy(funcName, "crop_mirror_normalize");
+        break;
+    case 7:
+        strcpy(funcName, "color_twist");
+        break;
+    }
+
+    if (ip_bitDepth == 0)
+    {
+        strcat(funcName, "_u8_");
+    }
+    else if (ip_bitDepth == 1)
+    {
+        strcat(funcName, "_f16_");
+    }
+    else if (ip_bitDepth == 2)
+    {
+        strcat(funcName, "_f32_");
+    }
+    else if (ip_bitDepth == 3)
+    {
+        strcat(funcName, "_u8_f16_");
+    }
+    else if (ip_bitDepth == 4)
+    {
+        strcat(funcName, "_u8_f32_");
+    }
+    else if (ip_bitDepth == 5)
+    {
+        strcat(funcName, "_i8_");
+    }
+    else if (ip_bitDepth == 6)
+    {
+        strcat(funcName, "_u8_i8_");
+    }
+
+    char func[1000];
+    strcpy(func, funcName);
+    strcat(func, funcType);
+    printf("\n\nRunning %s...", func);
+
+    int missingFuncFlag = 0;
+
+    int i = 0, j = 0;
+    int minHeight = 30000, minWidth = 30000, maxHeight = 0, maxWidth = 0;
+    int minDstHeight = 30000, minDstWidth = 30000, maxDstHeight = 0, maxDstWidth = 0;
+    unsigned long long count = 0;
+    unsigned long long ioBufferSize = 0;
+    unsigned long long oBufferSize = 0;
+    static int noOfImages = 0;
+
+    Mat image, image_second;
+
+    struct dirent *de;
+    char src1[1000];
+    strcpy(src1, src);
+    strcat(src1, "/");
+    char src1_second[1000];
+    strcpy(src1_second, src_second);
+    strcat(src1_second, "/");
+    strcat(funcName, funcType);
+    strcat(dst, "/");
+    strcat(dst, funcName);
+
+    DIR *dr = opendir(src);
+    while ((de = readdir(dr)) != NULL)
+    {
+        if (strcmp(de->d_name, ".") == 0 || strcmp(de->d_name, "..") == 0)
+            continue;
+        noOfImages += 1;
+    }
+    closedir(dr);
+
+    RppiSize *srcSize = (RppiSize *)calloc(noOfImages, sizeof(RppiSize));
+    RppiSize *dstSize = (RppiSize *)calloc(noOfImages, sizeof(RppiSize));
+    const int images = noOfImages;
+    char imageNames[images][1000];
+
+    DIR *dr1 = opendir(src);
+    while ((de = readdir(dr1)) != NULL)
+    {
+        if (strcmp(de->d_name, ".") == 0 || strcmp(de->d_name, "..") == 0)
+            continue;
+        strcpy(imageNames[count], de->d_name);
+        char temp[1000];
+        strcpy(temp, src1);
+        strcat(temp, imageNames[count]);
+        if (ip_channel == 3)
```

Comment:
Codacy found an issue: [The comparison 'ip_channel == 3' is always true.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/rpp/pullRequest?prid=5976402)